### DISCUSSION TITLE
Too many connections to Elasticsearch

### DISCUSSION THREAD

**Discussion by gpedras** [jertel/elastalert2#828] [Q&A] [answered]
Hi,
I'm having an issue where Elasticsearch is overwhelmed by Elastalert2 connections. I have 1457 rules but I can't seem to run Elastalert2 with that many rules unfortunately. Is there a way to make this work? 

I've tried many different configurations but the problem remains. Here's a sample of the errors:

```
Traceback (most recent call last):
  File "/home/xxxxxxx/.local/lib/python3.8/site-packages/elasticsearch-7.10.1-py3.8.egg/elasticsearch/connection/http_requests.py", line 161, in perform_request
  File "/home/xxxxxxx/.local/lib/python3.8/site-packages/requests-2.27.1-py3.8.egg/requests/sessions.py", line 645, in send
  File "/home/xxxxxxx/.local/lib/python3.8/site-packages/requests-2.27.1-py3.8.egg/requests/adapters.py", line 519, in send
requests.exceptions.ConnectionError: HTTPConnectionPool(host='es-host', port=9200): Max retries exceeded with url: /myindex*/_search?_source_includes=timestamp%2C%2A&ignore_unavailable=true&scroll=30s&size=10000 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f7500644640>: Failed to establish a new connection: [Errno 16] Device or resource busy'))

ERROR:elastalert:Error running query: ConnectionError(HTTPConnectionPool(host='es-host', port=9200): Max retries exceeded with url: /myindex*/_search?_source_includes=timestamp%2C%2A&ignore_unavailable=true&scroll=30s&size=10000 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f73db5e0160>: Failed to establish a new connection: [Errno 16] Device or resource busy'))) caused by: ConnectionError(HTTPConnectionPool(host='es-host', port=9200): Max retries exceeded with url: /myindex*/_search?_source_includes=timestamp%2C%2A&ignore_unavailable=true&scroll=30s&size=10000 (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f73db5e0160>: Failed to establish a new connection: [Errno 16] Device or resource busy')))
```

Here's a snippet of the configuration i'm using:

```
es_host: es-host
es_hosts:
        - es-host:9200
        - es-host2:9200
        - es-host3:9200
es_conn_timeout: 20000
es_port: 9200
```

I've increased the timeout but it doesn't seem to work either, Is there a way to limit the connections to Elasticsearch at a given time? Or a work-around?

Thanks you

EDIT1: misstyping

**Comment by jertel** [comment, marked as answer]
You might be running into the max open files limit. Some distro's set the default limit to 1024. You can adjust this fairly easily. Search Google for more information since the procedure varies between distros.

**Reply by gpedras** [reply]
> You might be running into the max open files limit. Some distro's set the default limit to 1024. You can adjust this fairly easily. Search Google for more information since the procedure varies between distros.

I have done that to every machine in the ES cluster, including the Elastalert2 machine, it doesn't solve the problem

**Reply by nsano-rururu** [reply]
@gpedras 

Why is there no `es-host` specified by `es_host` in the es_hosts specification? Are you looking at the documentation?

**Reply by gpedras** [reply]
> @gpedras
> 
> Why is there no `es-host` specified by `es_host` in the es_hosts specification? Are you looking at the documentation?

Misstyping, I changed the values when I made this post. Not relevant

**Reply by nsano-rururu** [reply]
Isn't it the same state even if the setting of es_hosts is deleted? I feel like there is a problem with es-host

**Reply by nsano-rururu** [reply]
You need to isolate the problem.
